src/app/tools/weak-passwords-report.component.ts

```diff
@@ -20,6 +20,15 @@ import { CipherReportComponent } from './cipher-report.component';
     templateUrl: 'weak-passwords-report.component.html',
 })
 export class WeakPasswordsReportComponent extends CipherReportComponent implements OnInit {
+
+    private static hasUsername(c: CipherView): boolean {
```

Comment:
Any reason these need to be static? I would think you could have made them local `const` functions within the `passwordStrengthCache()` method, OR at least instance local (not static).

Reply:
Instance local doesn't make sense because they don't touch anything instance local, and `passwordStrengthCache` is a `Map` specific to this class so I wanted those utility functions tied to this class.

Making them local const function within the `setCiphers()` method is acceptable, but we have to be careful because they are called in the closure of `allCiphers.forEach`. Probably I have to change either variable or function name of `hasUsername` not to capture the name.

Among those options I think making them static functions are better.

Reply:
> fitting greater body of code

In that sense I'd choose to make them local const function inside the `setCiphers()` method. I don't have time and environment to touch here right now so I'll push the fix of this branch next week.